components/style/custom_properties.rs

```diff
@@ -933,7 +808,7 @@ fn substitute_all(custom_properties_map: &mut CustomPropertiesMap, environment:
 
     // We have to clone the names so that we can mutably borrow the map
     // in the context we create for traversal.
-    let names = custom_properties_map.index.clone();
+    let names: Vec<_> = custom_properties_map.keys().cloned().collect();
     for name in names.into_iter() {
```

Comment:
Could we just iterate with `keys()` or maybe `keys().into_iter()` ?

```diff
- for name in names.into_iter() {
+ for name in custom_properties_map.keys() {
```

Reply:
`keys()` iterates over reference. 
`keys().cloned()` gives value but both keeps an immutable reference to `custom_properties_map` which prevents from borrowing it mutably at
```rust
            map: custom_properties_map,
```

Reply:
Oh, understood! But maybe we don't need to `collect()` it and then `into_iter()`? 🤔

Ex.

```rust
for name in custom_properties_map.keys().cloned() {
```

Reply:
If `cloned()` doesn't work, perhaps `map.keys().map(|name| name.clone())` would?

Reply:
`collect()` is fine for now. If we wanted to avoid the heap allocation we could collect into a `SmallVec` or such, but that should probably be a separate PR, and probably with some measurement, indication that it can actually help.